Realm/ObjectServerTests/RLMObjectServerTests.mm

```diff
@@ -1979,6 +1979,133 @@ - (void)testSyncConfigShouldNotMigrate {
+    [syncedRealm.syncSession suspend];
+    [syncedRealm transactionWithBlock:^{
+        [syncedRealm addObject:[Person ringo]];
+    }];
+    // Cannot export a synced realm as not all changes have been synced.
+    NSError *error;
+    [syncedRealm writeCopyWithConfiguration:syncConfig2 error:&error];
+    XCTAssertEqual(error.code, RLMErrorFail);
+    XCTAssertTrue([error.userInfo[NSLocalizedDescriptionKey] isEqualToString:@"Could not write file as not all client changes are integrated in server"]);
+}
+
+- (void)testWriteCopyWithConfigurationLocalRealmForSyncWithExistingData {
+    RLMUser *initialUser = [self userForTest:_cmd];
+    RLMRealmConfiguration *initialSyncConfig = [initialUser configurationWithPartitionValue:NSStringFromSelector(_cmd)];
+    initialSyncConfig.objectClasses = @[Person.class];
+
+    // Make sure objects with confliciting primary keys sync ok.
+    RLMObjectId *conflictingObjectId = [RLMObjectId objectId];
+    Person *person = [Person ringo];
+    person._id = conflictingObjectId;
+
+    RLMRealm *initialRealm = [RLMRealm realmWithConfiguration:initialSyncConfig error:nil];
+    [initialRealm transactionWithBlock:^{
+        [initialRealm addObject:person];
+        [initialRealm addObject:[Person john]];
+    }];
+    [self waitForUploadsForRealm:initialRealm];
+
+    RLMRealmConfiguration *localConfig = [RLMRealmConfiguration new];
+    localConfig.objectClasses = @[Person.class];
+    localConfig.fileURL = RLMTestRealmURL();
+
+    RLMUser *user = [self logInUserForCredentials:[self basicCredentialsWithName:@"SyncWithExistingDataUser"
+                                                                        register:YES]];
+    RLMRealmConfiguration *syncConfig = [user configurationWithPartitionValue:NSStringFromSelector(_cmd)];
+    syncConfig.objectClasses = @[Person.class];
+
+    RLMRealm *localRealm = [RLMRealm realmWithConfiguration:localConfig error:nil];
+    // `person2` will override what was previously stored on the server.
+    Person *person2 = [Person paul];
+    person2._id = conflictingObjectId;
+    person2.firstName = @"John";
+    person2.lastName = @"Doe";
+
+    [localRealm transactionWithBlock:^{
+        [localRealm addObject:person2];
+        [localRealm addObject:[Person george]];
+    }];
+
+    [localRealm writeCopyWithConfiguration:syncConfig error:nil];
+
+    RLMRealm *syncedRealm = [RLMRealm realmWithConfiguration:syncConfig error:nil];
+    [self waitForDownloadsForRealm:syncedRealm];
+    XCTAssertEqual([syncedRealm allObjects:@"Person"].count, 3);
+    [syncedRealm transactionWithBlock:^{
+        [syncedRealm addObject:[Person stuart]];
+    }];
+
+    [self waitForUploadsForRealm:syncedRealm];
+    RLMResults<Person *> *syncedResults = [Person allObjectsInRealm:syncedRealm];
+
+    NSPredicate *p = [NSPredicate predicateWithFormat:@"firstName = 'John' AND lastName = 'Doe' AND _id = %@", conflictingObjectId];
```

Comment:
Just wondering - should behavior be John Doe overwrite ringo, or should ringo stay over John Doe.
I don't know one way or another.